app/services/order_factory.rb

```diff
@@ -14,8 +14,22 @@ def create
     create_order
     set_user
     build_line_items
+
+    # Transition to :address state.
+    @order.next!
+
     set_addresses
+
+    # Transition to :delivery state.
+    @order.next!
+
+    set_shipping_method
+
+    # Transition to :payment state.
+    @order.next!
+
```

Comment:
you are assuming a specific order workflow here...

also, do you know what happens if one of these steps fails for some reason?
you could run @order.next! only if order.state is the expected state and you could do something if @order.next! fails

I am sorry I dont have a specific suggestion for this one...

Reply:
@luisramos0 What do you think about this approach: https://github.com/openfoodfoundation/openfoodnetwork/pull/3560/commits/7206cd88995ba52b77763a22d377c8564505a164 https://github.com/openfoodfoundation/openfoodnetwork/pull/3560/commits/dd2f7ecac65e3222afddd20748243fcfd8691c66

With these changes, shipment records are now created without changing the order status (similar to what is done for payments), and we now remember the previously selected shipping method and select it again after the new shipping rates have been created.